core.spec.md

```diff
@@ -0,0 +1,501 @@
+{
+  query: Query
+}
+
+type User {
+  # Specifying `as: "eg"` transforms @example into @eg
+  name: String @eg(data: ITEM)
+}
+
+# Additional specified schema elements must have their prefixes set
+# to the new name.
+#
+# This data enum was specified as `example__Data`, but will be renamed
+# as `eg__Data`:
+enum eg__Data {
+  ITEM
+}
+
+# Name transformation must also be applied to definitions pulled in from
+# specifications.
+directive @eg(data: eg__Data) on FIELD
+
+# (...other definitions omitted...)
+```
+
+###! export: Boolean! = false
+
+Control whether the entire feature is exported to the API.
+
+If {true}, all schema elements--including all directive usages--from this feature will be included in the schema's API *unless* they are annotated with {@core__export}`(export: false)`.
+
+If {false} (the default), no schema elements from this feature will be included in the schema's API *unless* they are annotated with {@core__export}.
+
+##! @core__export
+
+Specify whether the annotated element is exported to the API.
+
+```graphql definition
+directive @core__export(export: Boolean! = true)
+  repeatable on
+  | SCALAR
+  | OBJECT
+  | FIELD_DEFINITION
+  | ARGUMENT_DEFINITION
+  | INTERFACE
+  | UNION
+  | ENUM
+  | ENUM_VALUE
+  | INPUT_OBJECT
+  | INPUT_FIELD_DEFINITION
+```
+
+{@core__export} can occur at any type system location. Elements with {@core__export} will always be included in the API. Elements with {@core__export}`(export: false)` will always be excluded from the API.
+
+###! export: Boolean
+
+If true, the element is always exported, regardless of whether the feature which defines it [is exported](#@core/export). If false, the element is never exported.
+
+# Prefixing
+
+With the exception of a single root directive, core feature specifications MUST prefix all schema elements they introduce. The prefix:
+  1. MUST match the default name of the feature as derived from the feature's specification URL,
+  2. MUST be a string of characters valid within GraphQL names, and
+  3. MUST NOT contain the core namespace separator, which is two underscores ({"__"}).
+
+Prefixed names consist of the name of the feature, followed by two underscores, followed by the name of the element (which can be any valid GraphQL identifier). For instance, the `core` specification (which you are currently reading) introduces elements named [{@core}](#@core) [{@core__export}](#@core__export).
+
+A feature's *root directive* is an exception to the prefixing requirements. Feature specifications MAY introduce a single directive which carries only the name of the feature, with no prefix required. For example, the `core` specification introduces a {@core} directive. This directive has the same name as the feature ("`core`"), and so requires no prefix.
+
+```graphql example -- Using the @core directive with a spec's default name
+schema @core(feature: "https://spec.example.com/example/v1.0") {
```

Comment:
EXAMPLE: Should include the self-reference `@core` (and its definition or a comment saying we aren't)